api/python/quilt3/packages.py

```diff
@@ -225,6 +226,40 @@ def get_cached_path(self):
             return ObjectPathCache.get(physical_key)
         return None
 
+    def get_bytes(self, use_cache_if_available=True):
+        """
+        Returns the bytes of the object this entry corresponds to. If 'use_cache_if_available'=True, will first try to
+        retrieve the bytes from cache.
+        """
+        if use_cache_if_available:
+            cached_path = self.get_cached_path()
+            if cached_path is not None:
+                return get_bytes(pathlib.Path(cached_path).as_uri())
+
+        physical_key = _to_singleton(self.physical_keys)
+        data = get_bytes(physical_key)
+        return data
+
+    def get_as_json(self, use_cache_if_available=True):
+        """
+        Return the object as JSON. Assumes that the file is encoded using utf-8.
+
+        If 'use_cache_if_available'=True, will first try to retrieve the object from cache.
+        """
+        obj_bytes = self.get_bytes(use_cache_if_available=use_cache_if_available)
+        return json.loads(obj_bytes.decode("utf-8"))
+
+
+    def get_as_string(self, use_cache_if_available=True):
+        """
+        Return the object as a string. Assumes that the file is encoded using utf-8.
+
+        If 'use_cache_if_available'=True, will first try to retrieve the object from cache.
+        """
+        obj_bytes = self.get_bytes(use_cache_if_available=use_cache_if_available)
+        return obj_bytes.decode("utf-8")
```

Comment:
do we want `.decode("utf-8", "ignore")`? it's a bit more robust

Reply:
No, I think we generally want to fail loudly if the bytes are not valid utf-8